### PR TITLE
Unroll negative loop bounds and retain pragmas inside unrolled loop body

### PR DESCRIPTION
**Pull request by awnawab**
This PR provides the following two fixes to the loop unrolling functionality:
- Unroll loops with negative bounds
- Retain pragmas inside the unrolled loop body

### CODE REVIEW

**Comment by awnawab** [comment]
Can I please push an update here? I've thought of an edge case that would break the updates herein.

File: loki/transformations/transform_loop.py [hunk 1]
```diff
@@ -681,11 +679,14 @@ def visit_Loop(self, o, depth=None):
 
             return as_tuple(flatten(acc))
 
+        _pragma = o.pragma if o.pragma and not 'unroll' in o.pragma[0].content.lower() else None
+        _pragma_post = o.pragma_post if o.pragma_post and not 'unroll' in o.pragma_post[0].content.lower() else None
```
[the remark this answers is not in the document]
Comment (awnawab) [reply loki/transformations/transform_loop.py:683]: Thanks, that is indeed much safer! And yup, pragma/pragma_post is always a tuple.

**Review by reuterbal — APPROVED** [review]
Many thanks for taking care of the comment - and I like that you have even further cleaned-up my proposal via the use of pragma utilities!

Also, I should have applauded earlier on how super-clean the implementation: The use of `is_constant` and `get_pyrange` is really neat!

**Review by mlange05 — APPROVED** [review]
Looks good to me. :shipit:

**Pull request merged**